src/Microsoft.TemplateEngine.TemplateLocalizer/Commands/ModelBoundExecutableCommand.cs

```diff
@@ -12,18 +14,22 @@ namespace Microsoft.TemplateEngine.TemplateLocalizer.Commands
     /// <typeparam name="TModel">Type of the model class whose properties represent the arguments.</typeparam>
     internal abstract class ModelBoundExecutableCommand<TModel> : ExecutableCommand where TModel : class
     {
-        internal ModelBoundExecutableCommand(ILoggerFactory loggerFactory) : base(loggerFactory) { }
-
-        // @inheritdoc
-        public override Task<int> InvokeAsync(InvocationContext context)
+        internal ModelBoundExecutableCommand(string name, string? description = null, ILoggerFactory? loggerFactory = null)
+            : base(name, description, loggerFactory)
         {
-            return CommandHandler.Create<TModel, CancellationToken>(ExecuteAsync).InvokeAsync(context);
+            this.SetHandler((TModel model) => ExecuteAsync(model), GetModelBinder());
         }
 
         /// <summary>
         /// Executes the command with the given input.
         /// </summary>
         /// <param name="args">Arguments for the command.</param>
         protected abstract Task<int> ExecuteAsync(TModel args, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Gets the model binder for the command.
+        /// </summary>
+        /// <returns></returns>
+        protected abstract BinderBase<TModel> GetModelBinder();
```

Comment:
Since there is no auto model binding anymore and everyone will have to implement their own binding, I suggest we remove this type and move the contents into `ExecutableCommand` class.